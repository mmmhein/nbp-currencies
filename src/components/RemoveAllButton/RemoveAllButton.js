import React from 'react';
import PropTypes from 'prop-types';

import Button from '../Button';

function RemoveAllButton({ removeAllFollowedRates }) {
  const handleOnClick = () => removeAllFollowedRates();

  return (
    <Button grey onClick={handleOnClick}>
      Remove All
    </Button>
  );
}

RemoveAllButton.propTypes = {
  removeAllFollowedRates: PropTypes.func.isRequired
};

export default RemoveAllButton;
