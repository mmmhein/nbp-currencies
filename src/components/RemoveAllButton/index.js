import { connect } from 'react-redux';

import { removeAllFollowedRates } from '../../actions/followedRatesActions';
import RemoveAllButton from './RemoveAllButton';

function mapDispatchToProps(dispatch) {
  return {
    removeAllFollowedRates: () => dispatch(removeAllFollowedRates())
  };
}

export default connect(
  null,
  mapDispatchToProps
)(RemoveAllButton);
