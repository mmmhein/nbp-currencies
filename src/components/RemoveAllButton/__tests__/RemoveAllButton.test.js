import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';

import RemoveAllButton from '../RemoveAllButton';

afterEach(cleanup);

describe('COMPONENT - RemoveAllButton', () => {
  it('renders correctly', () => {
    const component = render(<RemoveAllButton removeAllFollowedRates={jest.fn()} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with grey background', () => {
    const component = render(<RemoveAllButton grey removeAllFollowedRates={jest.fn()} />);

    expect(component).toMatchSnapshot();
  });

  it('fire remove rates handler', () => {
    const removeAllFollowedRates = jest.fn();

    const { container } = render(<RemoveAllButton grey removeAllFollowedRates={removeAllFollowedRates} />);

    const button = container.querySelector('button');

    fireEvent.click(button);

    expect(removeAllFollowedRates).toHaveBeenCalled();
  });
});
