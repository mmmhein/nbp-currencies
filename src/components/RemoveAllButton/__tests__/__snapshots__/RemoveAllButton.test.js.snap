// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - RemoveAllButton renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  background: #666666;
  border: 0;
  color: #ffffff;
  display: block;
  font-size: 1.8rem;
  font-weight: 700;
  padding: 0.8rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  max-width: 15rem;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:focus {
  outline: 4px solid #ffbf47;
  outline-offset: 0;
}

@media (max-width:26.25em) {
  .c0 {
    font-size: 1.4rem;
    max-width: 9rem;
  }
}

<body>
    <div>
      <button
        class="c0"
      >
        Remove All
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-bdVaJa ItcSe"
    >
      Remove All
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`COMPONENT - RemoveAllButton renders correctly with grey background 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  background: #666666;
  border: 0;
  color: #ffffff;
  display: block;
  font-size: 1.8rem;
  font-weight: 700;
  padding: 0.8rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  max-width: 15rem;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:focus {
  outline: 4px solid #ffbf47;
  outline-offset: 0;
}

@media (max-width:26.25em) {
  .c0 {
    font-size: 1.4rem;
    max-width: 9rem;
  }
}

<div>
      <button
        class="c0"
      >
        Remove All
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-bdVaJa ItcSe"
    >
      Remove All
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
